src/translator/service.cpp

```diff
@@ -56,51 +56,49 @@ Service::translateMultiple(std::vector<std::string> &&inputs,
   // We queue the individual Requests so they get compiled at batches to be
   // efficiently translated.
   std::vector<std::future<Response>> responseFutures;
-  for (auto &input : inputs) {
-    std::future<Response> inputResponse =
-        queueRequest(std::move(input), responseOptions);
-    responseFutures.push_back(std::move(inputResponse));
+  for (size_t i = 0;  i < inputs.size(); i++) {
```

Comment:
I don't quite understand what is happening here. When translateMultiple is hit, you mimic the old behaviour using a callback under the hood?

Reply:
When is translateMultiple being used? How about a callback be executed on a vector of Responses, as opposed to this?

Reply:
The thing is @kpu has only a `blockIfWASM()`, while `--cpu-threads 0` previously supported a blocking workflow on even non-WASM. The bergamot-translator-app compiled here thus passed tests.

I have two options here:
1. Disable tests corresponding to blocking workflow on `--cpu-threads 0` on non-WASM. The code will work without issues.
2. Do some future trickery to get those tests to pass as well. This sequentially sets the promise and future in the `blockIfWASM()` case becoming moot.

Would you like me to go with 1? Only Mozilla uses `translateMultiple(...)`. This means the `std::promise/future` clutter here can go. I like to keep 2. It's a mess either way, because of a collapse of two different behaviours into the same class.

Reply:
@XapaJIaMnu's suggestions here require (i) negotiating with Mozilla to change signature (ii) Remove the adjustments changing brt, changing the run here also fighting even worse contraptions. 

Unless there are critical problems with going through an additional promise + future, I'm proposing to close this PR here. I'll be doing the test-adjustments restructuring things in a systematic way in a later PR if it's OK. 

All this PR does is change future to callback for https://github.com/browsermt/bergamot-translator/issues/126, which is complete.